service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/docmosis/AllOrdersDocmosisTest.java

```diff
@@ -0,0 +1,143 @@
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.docmosis.DocmosisHelper.extractPdfContent;
+import static uk.gov.hmcts.reform.fpl.docmosis.DocmosisHelper.remove;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readString;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocument;
+
+@SpringBootTest(classes = {
+    ObjectMapper.class,
+    OrderCreationService.class,
+    OrderDocumentGenerator.class,
+    ChildrenService.class,
+    OrderDocumentGenerator.class,
+    DocmosisDocumentGeneratorService.class,
+    OrderDocumentGeneratorHolder.class,
+    DocmosisCommonElementDecorator.class,
+    CaseDataExtractionService.class,
+    HearingVenueLookUpService.class,
+    DocumentMerger.class,
+    DocumentConversionService.class,
+    C21BlankOrderDocumentParameterGenerator.class,
+    C32CareOrderDocumentParameterGenerator.class,
+    C23EPODocumentParameterGenerator.class,
+    C35aSupervisionOrderDocumentParameterGenerator.class,
+    C47AAppointmentOfAChildrensGuardianParameterGenerator.class,
+    C33InterimCareOrderDocumentParameterGenerator.class,
+    C23EPOAdditionalDocumentsCollector.class,
+    DocmosisDocumentGeneratorService.class,
+    RestTemplate.class
+})
+
+public class AllOrdersDocmosisTest extends AbstractDocmosisTest {
+
+    private static final String LA_CODE = "LA_CODE";
+    private static final String LA_COURT = "La Court";
+
+    // TO implement before merge
+    private static final Set<Object> EXCLUDED_ORDERS = Set.of(
+        Order.C32_CARE_ORDER,
+        Order.C33_INTERIM_CARE_ORDER,
+        Order.C35A_SUPERVISION_ORDER,
+        Order.C47A_APPOINTMENT_OF_A_CHILDRENS_GUARDIAN
+    );
+
+    @SpyBean
+    private DocmosisDocumentGeneratorService generatorService;
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+    @MockBean
+    private DocumentDownloadService documentDownloadService;
+    @MockBean
+    private LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration;
+    @MockBean
+    private HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
+
+    private ResultsCaptor<DocmosisDocument> resultsCaptor = new ResultsCaptor<>();
+
+    @Autowired
+    private OrderCreationService underTest;
+
+    private final DocmosisOrderCaseDataGenerator dataGenerator = new DocmosisOrderCaseDataGenerator();
+
+    @ParameterizedTest
+    @MethodSource("allOrders")
+    void testAllOrders(Order order) {
+
+        String actualContent = getPdfContent(dataGenerator.generateForOrder(order));
+
+        assertThat(actualContent)
+            .isEqualToNormalizingWhitespace(readString(format("order-generation/%s.txt", order.name())));
+    }
+
+    private static Stream<Arguments> allOrders() {
+        return Stream.of(Order.values())
+            .filter(order -> !EXCLUDED_ORDERS.contains(order))
+            .map(Arguments::of);
+    }
+
+    public String getPdfContent(CaseData caseData, String... ignores) {
```

Comment:
this would also probs be worth while in the abstract class along with the results captor

Reply:
the mocking is specific for the class we are testing, not sure should go into the abstract superclass